python-frontend/src/test/java/org/sonar/python/semantic/v2/TypeInferenceV2Test.java

```diff
@@ -349,6 +351,25 @@ def foo(a, b): ...
     assertThat(secondCall.callee().typeV2()).isEqualTo(secondFunctionType);
   }
 
+  @Test
+  void child_class_method_call_is_not_a_member_of_parent_class() {
+    FileInput fileInput = inferTypes("""
+      class A:
+        def meth(self):
+          return self.foo()
+      class B(A):
+        def foo(self):
+          ...
+      """
+    );
+    Optional<ClassSymbol> classA = fileInput.globalVariables().stream().filter(s -> s.name().equals("A")).map(ClassSymbol.class::cast).findFirst();
```

Comment:
`ClassSymbol` is type inference V1. The assertions made in this test are essentially the same as in `TypeInferenceTest` (where they belong).

In `TypeInferenceV2Test`, we assert on the result of type inference V2, which means the result of `typeV2` calls that would provide an instance of `ClassType`.

That said, I guess we also could have the results of assertions on both `ClassSymbol` and `ClassType` to highlight the possible inconsistency (rather than looking both here and in the `TypeInferenceTest`), that's up to you.